sdk/js/src/Client.ts

```diff
@@ -43,10 +43,15 @@ export class DVCClient implements Client {
         this.store.saveUser(this.user)
             .then(() => console.log('Successfully saved user to local storage!'))
 
-        this.onInitialized = getConfigJson(environmentKey, this.user)
+        this.onInitialized = getConfigJson(environmentKey, this.user, options?.enableCloudData)
             .then((config) => {
                 const oldConfig = this.config
                 this.config = config as BucketedUserConfig
+
+                if (options?.enableCloudData || this.config.project.settings?.cloudEntityData?.enabled) {
```

Comment:
the logic here should be 
if enableCloudData is defined (ie. is a boolean) use its value
if not defined, use the value from the project settings